Rewrite & New features

AGuyNamedJens opened the pull request:
This is a draft for new features within the FactorioChatBot
Feel free to commit updates.

# Features:

- [x] Added support for nicknames from discord
- [x] Added deleteMessages option
- [x] Reformatted parseMessage
- [x] Prevent errors happening when sending messages
- [x] Add support to disable the sent message being returned by the bot
- [x] Add support for more information in the chat, such as when mentioning locations in-game
- [ ] Add support for more events such as `research_finished` if possible
- [x] Rewrite documentation from the readme to the wiki pages
- [ ] Optimize code

AGuyNamedJens committed:
Add deleteMessages option

AGuyNamedJens committed:
- Added support for nicknames from discord
- Added deleteMessages option
- Reformatted parseMessage
- Prevent errors happening when sending messages

AGuyNamedJens committed:
Make 'fancy' discord messages optional

Mattie112 commented:
Ah whoops I missed this! Feel free to close my MR in that case :)

AGuyNamedJens committed:
Optimization & rename sentMessage to sentMessages

AGuyNamedJens committed:
set deleteMessages to true by default

AGuyNamedJens committed:
Why did i think doubling all chat messages was a good idea?

mikhailmikhalchuk commented:
Definitely willing to contribute to this. Noticed some of the content on my fork on the main, so I'm assuming it got some traction. Have we considered switching the main language to TypeScript?

AGuyNamedJens commented:
@MikhailMCraft Hey, I never left this project behind, I just didn't get any new feature requests and i did indeed base some features and ideas from your fork.

I don't mind using typescript but wouldn't it make the "simple" bot, more advanced since building the TS files will be required?

I'm also considering automatic builds, which in this case would be the compiled files if we'll switch to typescript so the user won't have to compile everything on their own.

Let me know what you think.

mikhailmikhalchuk commented:
Cool. In the future, it's prudent for both parties usually to approach the author of the fork when merging code upstream. I do not personally mind (and the license allows it regardless), but it is something to keep in mind.

TypeScript files do not have to be prebuilt and can just be built on runtime. There is virtually no change from JavaScript in terms of runtime building as it is around the same speed.

If there is a Discord server, or you have a Discord handle, I'd be more than glad to continue discussing this on there.

AGuyNamedJens commented:
That's true. I might make a license change in the future to mention approaching the author when adding new features, would be a good idea to mention, wouldn't it?

Alright, in my own Discord bot project there's a huge difference between building and running directly, probably because it's larger than a 300-lined bot. We could indeed switch to typescript. Pre-builds would still probably be nice though.

I indeed have a discord server, feel free to join [here](https://jensbot.ovh/discord)

AGuyNamedJens committed:
Temporarily disable console commands
Add custom log file
Fix console messages appearently not logging properly to the console channel defined.
Added new listener for the custom events
Added ability to mute the Discord startup message

AGuyNamedJens committed:
Clear custom event log
Add custom log file to config.json.example

AGuyNamedJens committed:
Including instructions on how to start pm2 on server reboot ( #13 )

AGuyNamedJens committed:
Updates!
- Changed parseMessage to Switch cases
- Added GPS, Train (Stop) location support
- Added warning support
- Remove includes, replace with indication
